corehq/apps/reports/standard/inspect.py

```diff
@@ -86,7 +88,8 @@ def es_query(self):
                  .domain(self.domain)
                  .filter(time_filter(gte=self.datespan.startdate,
                                      lt=self.datespan.enddate_adjusted))
-                 .filter(self._get_users_filter(mobile_user_and_group_slugs)))
+                 .filter(self._get_users_filter(mobile_user_and_group_slugs)
+                         if mobile_user_and_group_slugs[0] else match_all()))
```

Comment:
I check the 0th index of `mobile_user_and_group_slugs` because when no filters are selected, it returns as a list with an empty string in it, which doesn't evaluate to False. Is is fair to assume this will happen in every case without filters selected?

Reply:
That's kind of odd, I'd expect an empty list, although maybe the URL is `...../?emwf=`, which is interpreted as an empty string.  I think I'd make this logic a `classmethod` something like `EMWF.all_users_selected(mobile_user_and_group_slugs)`, that the filter controls its own implementation.

Also, if this is to apply to all reports, then I think `user_es_query` will need to be modified as well.

Reply:
Oh, for the `mobile_user_and_group_slugs[0]` thing, I think you could use `any(mobile_user_and_group_slugs)` instead.